security/monitoring.adoc

```diff
@@ -0,0 +1,196 @@
+    "uid": "origin-node-1.example.local"
+  },
+  "kind": "Event",
+  "reason": "NodeHasDiskPressure",
+  ...
+}
+----
+
+Events related to resource creation, modification, or deletion can also be good candidates for detecting misuse of the cluster. The following query, for
+example, can be used to look for excessive pulling of images:
+
+[source, bash]
+----
+$ oc get events --all-namespaces -o json \
+  | jq '[.items[] | select(.involvedObject.kind == "Pod" and .reason == "Pulling")] | length'
+
+4
+----
+
+[NOTE]
+====
+When a namespace is deleted, its events are deleted as well. Events can also expire and are deleted to prevent filling up the *etcd* file. Events are
+not stored as a permanent record and frequent polling is necessary to capture statistics over time.
+====
+
+[[security-monitoring-logs]]
+== Cluster Logs
+This section describes the types of monitoring logs produced on the cluster.
+
+[[security-monitoring-service-logs]]
+=== Service Logs
+
+{product-title} produces logs with each *systemd* service that is running on a host:
+
+ifdef::openshift-origin[]
+- origin-master-api
+- origin-master-controllers
+- etcd
+- origin-node
+endif::[]
+ifdef::openshift-enterprise[]
+- atomic-openshift-master-api
+- atomic-openshift-master-controllers
+- etcd
+- atomic-openshift-node
+endif::[]
+
+These logs are intended for debugging purposes, not security auditing. They can be
+retrieved per host with `journalctl` or, in clusters with the aggregated logging stack deployed, in the logging _.operations_ indexes (possibly
+in the xref:../install_config/aggregate_logging.adoc#aggregated-ops[Ops cluster]) as a cluster administrator.
+
+[[security-monitoring-audit-log]]
+=== Master API Audit Log
+
+To log master API requests by users, administrators, or system components enable audit
+logging xref:../install_config/master_node_configuration.adoc#master-node-config-audit-config[for the master API]. This will
+create a file on each master host or, if there is no file configured, be included in the service's journal. Entries in the journal
+can be found by searching for *"AUDIT"*.
+
+xref:../install_config/master_node_configuration.adoc#master-node-config-audit-config[Audit
+log entries] consist of one line recording each REST request when it is
+received and one line with the HTTP response code when it completes. For
+example, here is a record of the system administrator requesting a list
+of nodes:
+
+[source, bash]
+----
+2017-10-17T13:12:17.635085787Z AUDIT: id="410eda6b-88d4-4491-87ff-394804ca69a1" ip="192.168.122.156" method="GET" user="system:admin" groups="\"system:cluster-admins\",\"system:authenticated\"" as="<self>" asgroups="<lookup>" namespace="<none>" uri="/api/v1/nodes"
+2017-10-17T13:12:17.636081056Z AUDIT: id="410eda6b-88d4-4491-87ff-394804ca69a1" response="200"
+----
+
+When requesting the list, it can be useful to poll the log periodically by using the following
+command to see if the number of requests failed:
```

Comment:
This doesn't make sense. Requesting what list? To see if the number what?

The example command shows the response codes seen in the last 5000 lines of the log and the number of responses that had each code. "Failure" isn't even clearly defined here. Auditors will have their own opinions about what is of concern.

Suggest: "It could be useful to poll the log periodically for the number of recent requests per response code, as in the following example:"